app/models/task.py

```diff
@@ -1,5 +1,46 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
+from datetime import datetime
+from typing import Optional
+from flask import make_response, abort
+from sqlalchemy import ForeignKey
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[Optional[datetime]]
+    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id")) 
+    goal: Mapped[Optional["Goal"]] = relationship(back_populates="tasks")
+
+
+    # book object to dict representation
+    def obj_to_dict(self):
+        task_as_dict = {}
+        task_as_dict["id"] = self.id
+        if self.goal:
+            task_as_dict["goal_id"] = self.goal.id
+        task_as_dict["title"] = self.title
+        task_as_dict["description"] = self.description
+        task_as_dict["is_complete"] = True if self.completed_at is not None else False
```

Comment:
Consider encapsulating this logic into a helper function, maybe something like `is_complete` that calculates what the value should be for the attribute and call it here.

```suggestion
        task_as_dict["is_complete"] = self.is_complete()
```